src/Modules/OrchardCore.Commerce/Controllers/PaymentController.cs

```diff
@@ -245,70 +223,44 @@ public async Task<IActionResult> SuccessPost(string orderId)
             await workflowManager.TriggerEventAsync(nameof(OrderCreatedEvent), order, "Order-" + order.ContentItemId);
         }
 
-        return Redirect($"~/success/{order.ContentItemId}");
-    }
+        var currentShoppingCart = await _shoppingCartPersistence.RetrieveAsync();
+        currentShoppingCart?.Items?.Clear();
 
-    [Route(nameof(ConfirmPayment))]
-    [HttpPost]
-    [ValidateAntiForgeryToken]
-    public async Task<IActionResult> ConfirmPayment(string paymentId)
-    {
-        PaymentIntent paymentIntent;
-
-        try
-        {
-            paymentIntent = await _stripePaymentService.GetPaymentIntentAsync(paymentId);
-        }
-        catch (StripeException exception)
-        {
-            return Json(new { error = exception.StripeError.Message });
-        }
-        catch (Exception exception)
-        {
-            _logger.LogError(exception, "Payment processing failed.");
-            var message = T["An error has occurred while processing the payment. Please verify and try again."];
-            return Json(new { error = message.Value });
-        }
+        // Shopping cart ID is null by default currently.
+        await _shoppingCartPersistence.StoreAsync(currentShoppingCart);
 
-        return await GeneratePaymentResponseAsync(paymentIntent);
+        // Set back to default, because a new payment intent should be created on the next checkout.
+        _paymentIntentPersistence.Store(paymentIntentId: string.Empty);
     }
 
-    private async Task<IActionResult> GeneratePaymentResponseAsync(PaymentIntent paymentIntent)
+    [AllowAnonymous]
+    [HttpGet("checkout/middleware")]
+    // Sadly that is how it is given.
+#pragma warning disable CA1707
```

Comment:
Include the warning message too, like it is with the suppression on line 58. But if you don't mind, please turn both into `[SuppressMessage]` attributes because that's cleaner for warnings in the method header.

Also you could use the Name property of `[FromQuery]` ([see](https://stackoverflow.com/a/38307063/14748624)) and keep the variable name camelCase.